0000-ilp-over-http.md

```diff
@@ -0,0 +1,57 @@
+---
+title: ILP Over HTTP
+draft: 1
+---
+
+# ILP Over HTTP
+> A bilateral communication protocol for server-to-server connections
+
+## Motivation
+
+Scaling Interledger infrastructure to handle large volumes of ILP packets requires horizontally scaling connectors. Using HTTP for bilateral communication enables service providers to leverage standard tools and services for hosting, load balancing, Distributed Denial of Service (DDoS) protection, and monitoring.
+
+## Overview
+
+In an ILP Over HTTP connection, both peers run HTTP servers with accessible HTTPS endpoints. When peering, the peers exchange their respective URLs, authentication tokens or TLS certificates, ILP addresses, and settlement-related details.
+
+Each ILP Prepare packet is sent as the body of an HTTP request to the peer's server endpoint. ILP Fulfill or Reject packets are returned as the body of the HTTP response.
+
+## Specification
+
+This is a minimal protocol built on HTTP. HTTP/2 is HIGHLY RECOMMENDED for performance reasons, although HTTP/1.1 MAY also be used. Implementations SHOULD support HTTP version negotiation via Application Protocol Negotiation (ALPN).
+
+### Authentication
+
+Peers MAY use any standard HTTP authentication mechanism to authenticate incoming requests. TLS Client Certificates are RECOMMENDED for security and performance.
```

Comment:
TLS client certs are a little tricky when the HTTP server in the plugin exists behind a load balancer (it's a plain HTTP server rather than HTTPS)

Reply:
I assume you can't even do client cert validation on a load balancer anyway. If you could (more likely on some kind of TLS appliance) you'd almost certainly expect the identity of the client to be passed along somehow, maybe in a header.

That said, writing a spec like this with strong assumptions about the deployment scenarios is a bad idea. The spec shouldn't assume this protocol will only be used by people behind a load balancer.

The reality is that client certs ARE better for performance so saying so is not wrong.